homeassistant/components/climate/tado.py

```diff
@@ -160,26 +203,40 @@ def current_temperature(self):
     @property
     def current_operation(self):
         """Return current readable operation mode."""
-        if self._cooling:
-            return "Cooling"
+        if self._current_operation in OPERATION_MANUAL:
+            if self._mode == CONST_MODE_COOL:
+                return OPERATION_MANUAL_COOL.get(self._current_operation)
+            if self._device_type == "HOT_WATER" and \
+                    not self._is_manual_temperature:
+                return OPERATION_MANUAL_HOT_WATER_ON.get(
+                    self._current_operation)
+            return OPERATION_MANUAL.get(self._current_operation)
         return OPERATION_LIST.get(self._current_operation)
 
     @property
     def operation_list(self):
         """Return the list of available operation modes (readable)."""
-        return list(OPERATION_LIST.values())
+        operations = list(OPERATION_LIST.values())
+        if CONST_MODE_COOL in self._min_temp:
+            operations.extend(list(OPERATION_MANUAL_COOL.values()))
+        elif CONST_MODE_HEAT in self._min_temp:
+            operations.extend(list(OPERATION_MANUAL.values()))
+        elif self._device_type == "HOT_WATER" and \
+                not self._is_manual_temperature:
+            operations.extend(list(OPERATION_MANUAL_HOT_WATER_ON.values()))
```

Comment:
This should not be in a climate component to begin with, there is a water heater component for this.

Reply:
At the current state off the hot_water component probably not... I would need to change the hot water device to get this working, like it is working right now.

Which would bring it closer to the climate device again which I’m not sure if desired. Also we will have the issue with the non default home-assistant modes.

@balloob what is the reason to split the hot_water? Because it seems to work tightly with your Heating since it’s the same system?